diff --git a/apt-mirror b/apt-mirror
index 7d9ff5c..9c2f54f 100755
--- a/apt-mirror
+++ b/apt-mirror
@@ -263,7 +263,7 @@ sub download_urls
 
         if ( $pid == 0 )
         {
-            exec 'wget', '--no-cache', '--limit-rate=' . get_variable("limit_rate"), '-t', '5', '-r', '-N', '-l', 'inf', '-o', get_variable("var_path") . "/$stage-log.$i", '-i', get_variable("var_path") . "/$stage-urls.$i", @args;
+            exec 'wget', '-nH', '--no-cache', '--limit-rate=' . get_variable("limit_rate"), '-t', '5', '-r', '-N', '-l', 'inf', '-o', get_variable("var_path") . "/$stage-log.$i", '-i', get_variable("var_path") . "/$stage-urls.$i", @args;
 
             # shouldn't reach this unless exec fails
             die("\n\nCould not run wget, please make sure its installed and in your path\n\n");
@@ -500,7 +500,7 @@ sub find_translation_files_in_release
     my ( $release_uri, $release_path, $line ) = '';
 
     $release_uri  = $dist_uri . "Release";
-    $release_path = get_variable("skel_path") . "/" . sanitise_uri($release_uri);
+    $release_path = get_variable("skel_path") . "/";
 
     unless ( open STREAM, "<$release_path" )
     {
@@ -556,7 +556,7 @@ sub process_translation_index
 
     $base_uri   = $dist_uri . $component . "/i18n/";
     $index_uri  = $base_uri . "Index";
-    $index_path = get_variable("skel_path") . "/" . sanitise_uri($index_uri);
+    $index_path = get_variable("skel_path") . "/";
 
     unless ( open STREAM, "<$index_path" )
     {
@@ -646,7 +646,7 @@ sub find_dep11_files_in_release
     my ( $release_uri, $release_path, $line ) = '';
 
     $release_uri  = $dist_uri . "Release";
-    $release_path = get_variable("skel_path") . "/" . sanitise_uri($release_uri);
+    $release_path = get_variable("skel_path") . "/";
 
     unless ( open STREAM, "<$release_path" )
     {
@@ -777,6 +777,7 @@ sub process_index
     my ( $path, $package, $mirror, $files ) = '';
 
     $path = sanitise_uri($uri);
+    $path =~ s&^[^/]+/&&;
     local $/ = "\n\n";
     $mirror = get_variable("mirror_path") . "/" . $path;
 
@@ -853,12 +854,12 @@ foreach (@config_sources)
         my $component;
         foreach $component (@components)
         {
-            process_index( $uri, "/dists/$distribution/$component/source/Sources" );
+            process_index( $uri, "dists/$distribution/$component/source/Sources" );
         }
     }
     else
     {
-        process_index( $uri, "/$distribution/Sources" );
+        process_index( $uri, "$distribution/Sources" );
     }
 }
 
@@ -871,12 +872,12 @@ foreach (@config_binaries)
         my $component;
         foreach $component (@components)
         {
-            process_index( $uri, "/dists/$distribution/$component/binary-$arch/Packages" );
+            process_index( $uri, "dists/$distribution/$component/binary-$arch/Packages" );
         }
     }
     else
     {
-        process_index( $uri, "/$distribution/Packages" );
+        process_index( $uri, "$distribution/Packages" );
     }
 }
 
@@ -930,10 +931,10 @@ sub copy_file
 foreach (@index_urls)
 {
     die("apt-mirror: invalid url in index_urls") unless s[^(\w+)://][];
-    copy_file( get_variable("skel_path") . "/" . sanitise_uri("$_"), get_variable("mirror_path") . "/" . sanitise_uri("$_") );
-    copy_file( get_variable("skel_path") . "/" . sanitise_uri("$_"), get_variable("mirror_path") . "/" . sanitise_uri("$_") ) if (s/\.gz$//);
-    copy_file( get_variable("skel_path") . "/" . sanitise_uri("$_"), get_variable("mirror_path") . "/" . sanitise_uri("$_") ) if (s/\.bz2$//);
-    copy_file( get_variable("skel_path") . "/" . sanitise_uri("$_"), get_variable("mirror_path") . "/" . sanitise_uri("$_") ) if (s/\.xz$//);
+    copy_file( get_variable("skel_path") . "/", get_variable("mirror_path") . "/" );
+    copy_file( get_variable("skel_path") . "/", get_variable("mirror_path") . "/" ) if (s/\.gz$//);
+    copy_file( get_variable("skel_path") . "/", get_variable("mirror_path") . "/" ) if (s/\.bz2$//);
+    copy_file( get_variable("skel_path") . "/", get_variable("mirror_path") . "/" ) if (s/\.xz$//);
 }
 
 ######################################################################################
