diff --git a/bin/common b/bin/common
index d1eb8f9..30098c9 100644
--- a/bin/common
+++ b/bin/common
@@ -209,9 +209,6 @@ log () {
 # to the address configured in MAILTO (if non-empty)
 error () {
     log "$@"
-    if [[ -n "${MAILTO}" ]]; then
-        echo "$@" | mail -e -s "[$PROGRAM@$(hostname -s)] ERROR [$$]" ${MAILTO}
-    fi
 }
 
 # run a hook
diff --git a/bin/ftpsync b/bin/ftpsync
index e3960fa..bbc82d5 100755
--- a/bin/ftpsync
+++ b/bin/ftpsync
@@ -1,6 +1,8 @@
 #! /bin/bash
 # No, we can not deal with sh alone.
 
+[[ $DEBUG = true ]] && set -x
+
 set -e
 set -u
 # ERR traps should be inherited from functions too. (And command
@@ -113,25 +115,8 @@ cleanup() {
     # all done. Mail the log, exit.
     log "Mirrorsync done";
 
-    if [[ -n ${MAILTO} ]]; then
-        # In case rsync had something on stderr
-        if [[ -s ${LOGDIR}/rsync-${NAME}.error ]]; then
-            mail -E -s "[${PROGRAM}@$(hostname -s)] ($$) rsync ERROR on $(date +"%Y.%m.%d-%H:%M:%S")" ${MAILTO} < "${LOGDIR}/rsync-${NAME}.error"
-        fi
-        if [[ ${ERRORSONLY} = false ]]; then
-            # And the normal log
-            MAILFILES="${LOG}"
-            if [[ ${FULLLOGS} = true ]]; then
-                # Someone wants full logs including rsync
-                MAILFILES="${MAILFILES} ${LOGDIR}/rsync-${NAME}.log"
-            fi
-            cat ${MAILFILES} | mail -E -s "[${PROGRAM}@$(hostname -s)] archive sync finished on $(date +"%Y.%m.%d-%H:%M:%S")" ${MAILTO}
-        fi
-    fi
-
     savelog "${LOGDIR}/rsync-${NAME}.log"
     savelog "${LOGDIR}/rsync-${NAME}.error"
-    savelog "$LOG" > /dev/null
 
     rm -f "${LOCK}"
 }
@@ -590,11 +575,11 @@ fi
 # this trap.  (We can not trap on EXIT, because that is called when the main script
 # exits. Which also happens when we background the mainroutine, ie. while we still
 # run!)
-trap cleanup ERR TERM HUP INT QUIT
+trap cleanup EXIT
 
 # Start log by redirecting stdout and stderr there and closing stdin
 mkdir -p "$LOGDIR"
-exec >"$LOG" 2>&1 <&-
+exec 2>&1 <&-
 log "Mirrorsync start"
 
 # Look who pushed us and note that in the log.
@@ -775,83 +760,3 @@ fi
 
 # Remove the Archive-Update-in-Progress file before we push our downstreams.
 rm -f "${LOCK}"
-
-# Check if there is a newer version of ftpsync. If so inform the admin, but not
-# more than once every third day.
-if [[ -r ${TO}/project/ftpsync/LATEST.VERSION ]]; then
-    LATEST=$(< "${TO}/project/ftpsync/LATEST.VERSION")
-    if ! [[ ${LATEST} =~ [0-9]+ ]]; then
-        LATEST=0
-    fi
-    if [[ ${LATEST} -gt ${VERSION} ]]; then
-        if [[ -n ${MAILTO} ]]; then
-            difference=0
-            if [[ -f ${LOGDIR}/ftpsync.newversion ]]; then
-                stamptime=$(< "${LOGDIR}/ftpsync.newversion")
-                unixtime=$(date +%s)
-                difference=$(( $unixtime - $stamptime ))
-            fi
-            if [[ ${difference} -ge 259200 ]]; then
-                # Only warn every third day
-                mail -s "[$(hostname -s)] Update for ftpsync available" ${MAILTO} <<EOF
-Hello admin,
-
-i found that there is a new version of me available.
-Me lonely ftpsync is currently version: ${VERSION}
-New release of myself is available as:  ${LATEST}
-
-Me, myself and I - and the Debian mirroradmins - would be very grateful
-if you could update me. You can find the latest version on your mirror,
-check $(hostname -s):${TO}/project/ftpsync/ftpsync-${LATEST}.tar.gz
-
-You can ensure the validity of that file by using sha512sum or md5sum
-against the available checksum files secured with a signature from the
-Debian FTPMaster signing key.
-
-EOF
-
-                date +%s > "${LOGDIR}/ftpsync.newversion"
-            fi
-        fi
-    else
-        # Remove a possible stampfile
-        rm -f "${LOGDIR}/ftpsync.newversion"
-    fi
-fi
-
-if [[ ${HUB} = true ]]; then
-    # Trigger slave mirrors if we had a push for stage2 or all, or if its mhop
-    if [[ true = ${SYNCSTAGE2} ]] || [[ true = ${SYNCALL} ]] || [[ true = ${SYNCMHOP} ]]; then
-        RUNMIRRORARGS=""
-        if [[ -n ${ARCHIVE} ]]; then
-            # We tell runmirrors about the archive we are running on.
-            RUNMIRRORARGS="-a ${ARCHIVE}"
-        fi
-        # We also tell runmirrors that we are running it from within ftpsync, so it can change
-        # the way it works with mhop based on that.
-        RUNMIRRORARGS="${RUNMIRRORARGS} -f"
-
-        if [[ true = ${SYNCSTAGE1} ]]; then
-            # This is true when we have a mhop sync. A normal multi-stage push sending stage1 will
-            # not get to this point.
-            # So if that happens, tell runmirrors we are doing mhop
-            RUNMIRRORARGS="${RUNMIRRORARGS} -k mhop"
-        elif [[ true = ${SYNCSTAGE2} ]]; then
-            RUNMIRRORARGS="${RUNMIRRORARGS} -k stage2"
-        elif [[ true = ${SYNCALL} ]]; then
-            RUNMIRRORARGS="${RUNMIRRORARGS} -k all"
-        fi
-        log "Trigger slave mirrors using ${RUNMIRRORARGS}"
-        ${BASEDIR}/bin/runmirrors ${RUNMIRRORARGS}
-        log "Trigger slave done"
-
-        HOOK=(
-            HOOKNR=5
-            HOOKSCR=${HOOK5}
-        )
-        hook $HOOK
-    fi
-fi
-
-# All done, lets call cleanup
-cleanup
