diff --git a/Library/Homebrew/os/mac.rb b/Library/Homebrew/os/mac.rb
index 3bad6bb..144bc06 100644
--- a/Library/Homebrew/os/mac.rb
+++ b/Library/Homebrew/os/mac.rb
@@ -18,7 +18,7 @@ module OS
     # This can be compared to numerics, strings, or symbols
     # using the standard Ruby Comparable methods.
     def version
-      return @version ||= Version.new("0") unless OS.mac?
+      return @version ||= Version.new("10.12") unless OS.mac?
       @version ||= Version.new(full_version.to_s[/10\.\d+/])
     end
 
